### DISCUSSION TITLE
Is there a way to invoke dolos-lib from c#/.net API.

### DISCUSSION THREAD

**Discussion by sourabh-tiwari-gslab** [dodona-edu/dolos#971] [Q&A] [answered]
Is there a way to invoke dolos-lib from c#/.net API.

**Comment by rien** [comment, marked as answer]
There is no official way, but looks like you could be able to call NodeJS libraries from C# using this project: https://github.com/JeringTech/Javascript.NodeJS

**Comment by rien** [comment]
Can I ask what you are building and how you are planning to integrate Dolos? Is there anything we can provide with the API of the library that would help you?

**Reply by sourabh-tiwari-gslab** [reply]
We have a .net API that accepts a certain parameter that includes codes to check for the plagiarism of multiple users. 
Currently, with the Option, we have We were planning to host a Node API that will accept files and will invoke dolos-lib to call analyze files method.

But hosting another API just to call dolos is not something we are very much in favor of.It will be great If I could initialize the dolos library and pass files in c# directly.

**Reply by rien** [reply]
Then the project in my other answer would be a good solution I think (if it works, I haven't tried it).

As an alternative, we're currently writing a web server with an HTTP  API that is able to analyze files and host the results. This project is nearing its public release, would that help you?
